packages/ckeditor5-typing/src/deleteobserver.js

```diff
@@ -35,6 +35,20 @@ export default class DeleteObserver extends Observer {
 		} );
 
 		document.on( 'keydown', ( evt, data ) => {
+			// Do not fire the `delete` event on Windows, if Shift + Delete key combination on a non-collapsed selection was pressed.
+			//
+			// The Shift + Delete key combination should work in the same way as the `cut` event on a non-collapsed selection on Windows.
+			// In fact, the native `cut` event is actually emitted in this case, but with lower priority. Therefore, in order to handle the
+			// Shift + Delete key combination correctly, it is enough not to emit the `delete` event and stop the original `keydown` event,
+			// so that {@link module:typing/utils/injectunsafekeystrokeshandling `injectUnsafeKeystrokesHandling()`} will not interfere.
+			//
+			// See https://github.com/ckeditor/ckeditor5/issues/9326.
+			if ( env.isWindows && this._isShiftDelete( data ) && this._isNonCollapsedSelection( data ) ) {
+				evt.stop();
+
+				return;
+			}
```

Comment:
Please extract this code from the `DeleteObserver` to some listener for `delete` event outside this file. Remember that events fired by view Document are the bubbling ones so you probably will need to register that handler at the `context: $capture` to make sure that you stop that event before it starts bubbling.

Reply:
I extracted this code to the `Delete` plugin, because similar (OS-related) code already exists there. And it looks like I do not need to listen in the `$capture` phase.

Reply:
We wanted to avoid firing the `#delete` event when some conditions are met.

@niegowski, could you explain to us why the first proposed solution is invalid?

Reply:
I agree with @pomek. The original approach was to not fire `delete` when... no deletion happens. The current code works backward – it fixes what should not happen in the first place. So, I'm for reverting the change.

Reply:
Sorry for not explaining. My biggest concern is that that code should not be inside the observer, those are expected to translate events from the DOM to the view space. Also the `evt.stop()` is not consistent because of that change. For the case whether it should be on the delete event or the keydown event I was considering the `sequence` handling of the delete event but maybe it would be better to listen to keydown and stop it before even the delete observer could look at it. Or maybe it should all be inside the observer but the `evt.stop()` should be consistent? Now I'm not sure

Reply:
We have two types of observers – ones translating DOM->view events. And one giving more semantics to view events (keydown -> delete). 

I think that, in this case, the `keydown` is still a valid event (as keys were indeed pressed) and should not be stopped. I didn't check previously that @psmyrek actually stopped it so I take my previous comment back – the original approach wasn't good.

The second point is that `DeleteObserver` must not affect `keydown` because it's not its responsibility. `DeleteObserver`'s responsibility is to translate `keydown` to `delete` when `keydown` represents a deletion. And this is the only thing that should happen here.

@psmyrek is **not firing** `delete` enough to resolve this issue? If not, then the rest of the fix should, AFAICS, land in other places. E.g. I remember that we talked about "unsafe keystroke handler" and probably that the second part should be there.

Reply:
Thanks for clarification. I extracted the logic to a separate util `isShiftDeleteOnNonCollapsedSelection()` because now it has to be used in 2 places: in `DeleteObserver` and in `injectUnsafeKeystrokesHandling()` function.